Changes: synchronize trunk with the 4.05 branch

gasche opened the pull request.

gasche committed:
Changes: synchronize with the 4.05 branch

gasche commented:
To this Changes-train I added a minor commit to fix the change entry for #1012.

gasche committed:
[minor] Changes: GPR#1012 is not compatibility-breaking

Octachron commented on Changes:
```diff
@@ -24,26 +24,26 @@ Working version
 
-* GPR#659: Remove support for SPARC native code generation
-  (Mark Shinwell)
+- PR#7447, GPR#995: incorrect code generation for nested recursive bindings
```
PR#7447 ⇒ MPR#7447?

Octachron commented on Changes:
```diff
@@ -728,119 +753,109 @@ The complete list of changes is listed below.
+- PR#7511, GPR#1133: Unboxed type with unboxed argument should not be accepted
+  (Damien Doligez, review by Jeremy Yallop and Leo White)
+
+- GPR#795: remove 256-character limitation on Sys.executable_name
```
Duplicata at line 335?

Octachron reviewed (commented):
I found two minor points that could be fixed.

gasche commented:
@Octachron thanks, I fixed both issues. In fact I converted all PR# mentions in trunk into MPR#, because now that GPR# are four-digit numbers as well I think there are more opportunities for confusion.

gasche committed:
Changes: consistently use MPR# instead of PR#

gasche committed:
Changes: remove duplicate GPR#795 entry

(spotted by Octachron)

damiendoligez commented on Changes:
```diff
@@ -24,26 +24,26 @@ Working version
 
 - MPR#7501, GPR#1089: Consider arrays of length zero as constants
-- PR#7501, GPR#1089: Consider arrays of length zero as constants
+- MPR#7501, GPR#1089: Consider arrays of length zero as constants
```
The header line is duplicated, and the whole entry is duplicated from 4.04.1.

gasche replied:
Thanks, fixed. (I think these issues come from a weird git fuzzy merge glitch, at least not from the 4.05 version.). I kept the change in trunk and removed the one in 4.04.1 -- I don't think this change was even available in 4.05.

damiendoligez reviewed (approved):
LGTM except for the duplicated entry.
